Provide consistent examples of service invocation for all SDKs (#1965)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md

This article describe how to deploy services each with an unique application ID, so that other services can discover and call endpoints on them using service invocation API.

## Example:

> There are two services called order processing service and checkout service. Dapr used in order processing service is used to invoke checkout service. Both order processing service and checkout service have dapr side cars.

Reviewer: ```suggestion
The below code examples loosely describe an application that processes orders. In the examples, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars and the order processing service uses Dapr to invoke the checkout method in the checkout service.
```

Reviewer: @amulyavarote - Reworded this a bit. Thoughts?

Author: Looks good!

> The below code examples loosely describe an application that processes orders. In the examples, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars and the order processing service uses Dapr to invoke the checkout method in the checkout service.

Reviewer: ```suggestion
The below code examples loosely describe an application architecture that processes orders. In the examples, there are two services - an order processing service and a checkout service. Each service has a Dapr sidecar and the order processing service uses Dapr to invoke the checkout method in the checkout service.
```

dapr run --app-id orderprocessing --app-port 6001 --dapr-http-port 3601 --dapr-grpc-port 60001 go run OrderProcessingService.go

```

If your app uses an SSL connection, you can tell Dapr to invoke your app over an insecure SSL connection:

```bash

dapr run --app-id checkout --app-port 6002 --dapr-http-port 3602 --dapr-grpc-port 60002 --app-ssl go run CheckoutService.go

dapr run --app-id orderprocessing --app-port 6001 --dapr-http-port 3601 --dapr-grpc-port 60001 --app-ssl go run OrderProcessingService.go

```

{{% /codetab %}}

{{% codetab %}}

```bash

dapr run --app-id checkout --app-port 6002 --dapr-http-port 3602 --dapr-grpc-port 60002 npm start

dapr run --app-id orderprocessing --app-port 6001 --dapr-http-port 3601 --dapr-grpc-port 60001 npm start

```

If your app uses an SSL connection, you can tell Dapr to invoke your app over an insecure SSL connection:

```bash

dapr run --app-id checkout --app-port 6002 --dapr-http-port 3602 --dapr-grpc-port 60002 --app-ssl npm start

dapr run --app-id orderprocessing --app-port 6001 --dapr-http-port 3601 --dapr-grpc-port 60001 --app-ssl npm start

```

{{% /codetab %}}

{{% codetab %}}

> ### Setup an ID using Kubernetes

Reviewer 2: ```suggestion
### Set an app-id when deploying to Kubernetes
```

{{< /tabs >}}

## Step 2: Invoke the service

> Dapr uses a sidecar, decentralized architecture. To invoke an application using Dapr, you can use the `invoke` API on any Dapr instance.

Reviewer: ```suggestion
To invoke an application using Dapr, you can use the `invoke` API on any Dapr instance.
```

> The sidecar programming model encourages each applications to talk to its own instance of Dapr. The Dapr instances discover and communicate with one another.

Reviewer: ```suggestion
The sidecar programming model encourages each application to interact with its own instance of Dapr. The Dapr sidecars discover and communicate with one another.
```

> Below are the example code snippets leveraging Dapr SDKs.

Reviewer: ```suggestion
Below are code examples that leverage Dapr SDKs for service invocation.
```

> {{< tabs Dotnet Java Python Go Javascript curl CLI >}}

Reviewer: @amulyavarote is php not supported here?

Author: I have php code. But I have not executed it yet. Thats the reason I have not included it in the document.

{{% codetab %}}

> ```bash

Reviewer: if you change this to "dotnet", you will get syntax highlighting for dotnet. 

Same goes for the other languages

//headers

using Dapr.Client;
using System.Net.Http;

//code

CancellationTokenSource source = new CancellationTokenSource();
CancellationToken cancellationToken = source.Token;
using var client = new DaprClientBuilder().Build();
var result = client.CreateInvokeMethodRequest(HttpMethod.Get, "checkout", "checkout/" + orderId, cancellationToken);
await client.InvokeMethodAsync(result);

```
{{% /codetab %}}

{{% codetab %}}

> ```bash

Reviewer: if you change bash to java, you will get java syntax highlighting....same goes for all the rest of the languages

    null,
    HttpExtension.GET,
    String.class
);

```
{{% /codetab %}}

{{% codetab %}}
```bash

//headers

from dapr.clients import DaprClient

//code

with DaprClient() as daprClient:
  result = daprClient.invoke_method(
      "checkout",
          f"checkout/{orderId}",
          data=b'',
          http_verb="GET"
  )    

```
{{% /codetab %}}

{{% codetab %}}
```bash

//headers
import (
	dapr "github.com/dapr/go-sdk/client"
)

//code

client, err := dapr.NewClient()
if err != nil {
  panic(err)
}
defer client.Close()
ctx := context.Background()

result, err := client.InvokeMethod(ctx, "checkout", "checkout/" + strconv.Itoa(orderId), "get") 

```
{{% /codetab %}}

{{% codetab %}}
```bash

//headers

import { DaprClient, HttpMethod, CommunicationProtocolEnum } from 'dapr-client'; 

//code

const daprHost = "127.0.0.1"; 
const client = new DaprClient(daprHost, process.env.DAPR_HTTP_PORT, CommunicationProtocolEnum.HTTP);
const result = await client.invoker.invoke('checkout' , "checkout/" + orderId , HttpMethod.GET);   

```
{{% /codetab %}}

{{% codetab %}}
To invoke a 'GET' endpoint:

```bash
curl http://localhost:3602/v1.0/invoke/checkout/method/checkout/100
```

> ### Additional URL formats

Reviewer: We should not include HTTP proxy in the curl example. That section should stand by itself as an option for customers looking to adopt Dapr coming from an application that already has http client code calling existing endpoints and they do not want to change over to using the invoke url format.

My recommendation is to leave the "Additional URL formats" section outside of the code blocks. Moving forward, we should elaborate on this feature
